tests/rpc.js

```diff
@@ -0,0 +1,27 @@
+const test = require('tape')
+const sinon = require('sinon')
+const Manager = require('../lib/rpc/rpc-manager')
+
+test('eth_getBlockByNumber', t => {
+  const block = {
+    toJSON: sinon.stub().returns({})
+  }
+  const blockchain = {
+    getBlock: sinon.stub().yields(null, block)
+  }
+  const manager = new Manager(blockchain)
+  const req = {
+    jsonrpc: '2.0',
+    method: 'eth_getBlockByNumber',
+    params: ['0x1', true],
+    id: 1
+  }
+
+  t.false(blockchain.getBlock.called)
+  manager.execute(req, (err, block) => {
+    t.error(err)
+    t.true(blockchain.getBlock.called)
+    t.equal(blockchain.getBlock.firstCall.args[0], 1)
+    t.end()
+  })
+})
```

Comment:
So great that you already added some framing for test execution, really like this ``sinon`` library, didn't know that before.

I think our general reference for RPC execution should  be Geth behavior, this is also really practical to test running the JS client and Geth with RPC and different port side by side and then have a look at the results.

I've done this with JS ``8545`` and ``Geth`` ``8546`` and get divergent behavior e.g. for:

```shell
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1", true],"id":1}' http://127.0.0.1:8545
{...JSON Result...}
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1", true],"id":1}' http://127.0.0.1:8546
invalid content type, only application/json is supported
```

So Geth requires a content header ``-H "Content-Type: application/json"`` which I think we should then require as well.

Also things like leading zeros (0x01) differ:
```shell
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x01", true],"id":1}' -H "Content-Type: application/json" http://127.0.0.1:8545|jq
{...JSON Result...}
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x01", true],"id":1}' -H "Content-Type: application/json" http://127.0.0.1:8546|jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   197  100   117  100    80  73079  49968 --:--:-- --:--:-- --:--:--  114k
{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "code": -32602,
    "message": "invalid argument 0: hex number with leading zero digits"
  }
}
```

Can you test some basic edge cases (e.g. not-existing/downloaded block) and see how things behave and then adopt the code base to make sure that the output is the same?

Reply:
Hi @yurenju, I would probably merge https://github.com/ethereumjs/ethereumjs-client/pull/29 before this PR and then ask you to rebase on top, if you see any problems here let me know.

Reply:
@holgerd77 sure go ahead! I may update my pr in two days when I get more free time :D, will rebase after #29 is landed.

Reply:
and thanks for testing with js client and geth side by side, I will add some basic edge cases!

Reply:
I feel the way I handled rpc is too simple since we need to fit JSON-RPC spec. there are few Node.js packages for JSON-RPC and seems jayson is a well-maintain solution.

so on next commit, I will introduce jayson.

https://github.com/tedeh/jayson